00_Base/src/datasources/TariffsDatasource.ts

```diff
@@ -0,0 +1,115 @@
+import { Service } from 'typedi';
+import { OcpiTariffRepository } from '../repository/OcpiTariffRepository';
+import { OcpiTariff, TariffKey } from '../model/OcpiTariff';
+import { SequelizeTariffRepository, Tariff } from '@citrineos/data';
+import { TariffMapper } from '../mapper/tariff.mapper';
+import { TariffDTO } from '../model/DTO/tariffs/TariffDTO';
+import { Op } from 'sequelize';
+import { ILogObj, Logger } from 'tslog';
+import { PutTariffRequest } from '../model/DTO/tariffs/PutTariffRequest';
+import { OcpiSequelizeInstance } from '../util/sequelize';
+import { GetTariffsParams } from '../model/DTO/tariffs/GetTariffsParams';
+import { ITariffsDatasource } from './ITariffsDatasource';
+
+@Service()
+export class TariffsDatasource implements ITariffsDatasource {
+  constructor(
+    private ocpiTariffRepository: OcpiTariffRepository,
+    private coreTariffRepository: SequelizeTariffRepository,
+    private tariffMapper: TariffMapper,
+    private ocpiSequelizeInstance: OcpiSequelizeInstance,
+    private readonly logger: Logger<ILogObj>,
+  ) {
+  }
+
+  async getTariffByKey(key: TariffKey, isCoreTariffKey: boolean = false): Promise<TariffDTO | undefined> {
+    try {
+      const ocpiTariff = isCoreTariffKey
+        ? await this.ocpiTariffRepository.findByCoreTariffKey(key)
+        : await this.ocpiTariffRepository.findByTariffKey(key);
+
+      if (!ocpiTariff) return undefined;
```

Comment:
linter error: you have to wrap the if in curly braces, no shortcuts allowed i guess 😬